Neos.Cache/Classes/Backend/SimpleFileBackend.php

```diff
@@ -226,31 +254,16 @@ public function remove($entryIdentifier)
         }
         $cacheEntryPathAndFilename = $this->cacheDirectory . $entryIdentifier . $this->cacheEntryFileExtension;
         for ($i = 0; $i < 3; $i++) {
-            $result = false;
             try {
-                if (strncasecmp(PHP_OS, 'WIN', 3) == 0) {
-                    // On Windows, unlinking a locked/opened file will not work, so we just attempt the delete straight away.
-                    // In the worst case, the unlink will just fail due to concurrent access and the caller needs to deal with that.
-                    $result = unlink($cacheEntryPathAndFilename);
-                } else {
-                    $file = fopen($cacheEntryPathAndFilename, 'rb');
-                    if ($file === false) {
-                        continue;
-                    }
-                    if (flock($file, LOCK_EX) !== false) {
-                        $result = unlink($cacheEntryPathAndFilename);
-                        flock($file, LOCK_UN);
-                    }
-                    fclose($file);
-                }
+                $result = $this->tryRemoveWithLock($cacheEntryPathAndFilename);
 
                 if ($result === true) {
                     clearstatcache(true, $cacheEntryPathAndFilename);
                     return $result;
                 }
             } catch (\Exception $e) {
             }
-            usleep(rand(10, 500));
+            usleep(random_int(10, 500));
```

Comment:
IMHO this is not a good idea because random_int is slower than rand and no cryptographically secure value is needed here.